app/controllers/UserController.java

```diff
@@ -3,68 +3,64 @@
 /**
  * Created by Ejub on 31.1.2016.
  */
-import play.api.libs.json.JsPath;
+
+
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import models.User;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import play.libs.Json;
 import play.mvc.*;
+import utilities.Resources;
+import utilities.UserHelper;
 import views.html.*;
 import utilities.Email;
-import play.api.libs.json.Json;
-import play.api.libs.*;
-import play.data.validation.Validation;
-import play.data.validation.ValidationError;
-import play.data.validation.Constraints;
-import play.api.data.validation.*;
-
 import com.fasterxml.jackson.databind.JsonNode;
 import play.mvc.BodyParser;
+import utilities.Resources;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import org.joda.time.*;
+
+import java.security.SecureRandom;
 
-import models.Users;
 public class UserController extends Controller {
     /**
      * Registers the user
-     * @return
+     * @return the authorization token in JSON format
      */
     @BodyParser.Of(BodyParser.Json.class)
-    public Result register(String jsonInput) {
-
-        JsonNode json2 = request().body().asJson();
-        String name = json2.findPath("name").textValue();
-        String email = json2.findPath("email").textValue();
-        String surname = json2.findPath("surname").textValue();
-        String password = json2.findPath("password").textValue();
-        boolean isConfirmed = false;
-
-        Users user = new Users();
-        user.email = email;
-        user.name = name;
-        user.surname = surname;
-        user.password = password;
-
-        user.save();
-/*
-        if(name == null) {
-            return badRequest("Missing parameter [name]");
-        } else {
-            return ok("Hello " + name);
-        }
-*/
-        //Email.sendConfirmationEmail("ejubkadric@gmail.com", "link");
-
-        //response().setContentType("application/json");
-        return ok(index.render(name));
-    }
+    public Result register() {
+        JsonNode json = Json.parse(request().body().asJson().asText());
+        UserHelper userHelper = Json.fromJson(json, UserHelper.class);
 
-    /**
-     * Function sends confirmation mail with activation link using Email utility class
-     * @param emailAddress email of user
-     */
-    private void sendConfirmationMail(String emailAddress){
+        if(Resources.passwordCheck(userHelper))
+            return badRequest(index.render(Resources.BAD_REQUEST_PASSWORD_MISMATCH));
+
+        userHelper.createUser();
 
-        Email.sendConfirmationEmail(emailAddress, "blabla");
+        return ok(index.render(userHelper.getToken()));
     }
 
-    public Result test(String t){
 
-        return ok(index.render(t+" is the message"));
+    public Result confirm(String registrationToken){
+
+        UserHelper userHelper = new UserHelper();
```

Comment:
Don't use UserHelper for mapping request data. Use persistent models directly instead (class User, in this case). You can have helper/utility classes for grouping some general stuff (like token generation, password encryption, etc), but not as a copy of models. You can mark attributes that should be ignored for JSON parsing (explore @JsonIgnore annotation), as well as the ones that should be ignored during persistence (explore @Transient annotation).